### PR TITLE
Allowed IP as alternative auth

### PR DESCRIPTION
**Pull request by serk7**
Actually i'm working with a project with same name called CloudProxy ([https://github.com/NoahCardoza/CloudProxy](https://github.com/NoahCardoza/CloudProxy)). It's to pass throw cloudflare challenge. To do that is using a chrome with pupeeter. It's starting a chrome browser passing a proxy via parameter but it doesn't accept user and password on it ([https://superuser.com/questions/902620/google-chrome-proxy-settings-with-username-and-password](https://superuser.com/questions/902620/google-chrome-proxy-settings-with-username-and-password)).
This is the motivation to fork it and implement an alternative way to authenticate with proxy than using user and password. I used the parameter allowed_ip from tinyproxy.

Implemented:
- Alternatrive authentication using ALLOWED_IP
- Variable environments that are using boolean now are using real boolean instead of String. Instead of using ENABLE_AWS=True now you should write ENABLE_AWS=true. And in code we are using now (if ENABLE_AWS:) instead of (if ENABLE_AWS='True')
- Check if proxy is working we are trying to load google.com throw the proxy instead of check the proxy url directly. For some strange reason proxy server response 403 HTTP forbidden when you visit is directly. Ex: http://10.12.23.3:8899 if it's using allowed_id, but it's really working as proxy. Increased timeout to 6 to let the proxy do its job.
- Added optional parameter AWS_KEY_NAME to pass throw  a pairkey to login on the EC2 instance.
- Added optional parameter PROXY_STEALTH=true, default false to set proxy in stealth mode and not sending proxy headers on its requests to the sites.
- Update README.md and docs/aws.md

Since is the first time i do something on python and docker. Probably is not really optimitzed and i guess there is space to enhance. I tested it on DigitalOcean, Hetzner and AWS and it works flawless without problems atm. 

I published a docker image: bubexel/cloudprroxy:latest

Thank you for your work on it!

Greetings

### CODE REVIEW

**Comment by claffin** [comment]
Many thanks for your contribution and sorry for the late PR review. Taking a look at this over the next day or so to merge.

File: cloudproxy/providers/config.py [hunk 1]
```diff
@@ -13,5 +13,5 @@ def set_auth(username, password, allowed_ip):
         else:
             filedata = filedata.replace("BasicAuth username password", "#BasicAuth username password")
         if len(allowed_ip) > 0:
-            filedata = filedata.replace("#Allow 127.0.0.1", "Allow " + allowed_ip)
+            filedata = filedata.replace("#Allow 127.0.0.1", "Allow " + allowed_ip + "\/24")
```
Comment (claffin) [inline cloudproxy/providers/config.py:16]: My understanding here is your adding /24 to the end of the allowed_ip which would mean the entire subnet has access. Not sure this is very good idea and not sure the reasoning behind it. A total of 256 IPs would then have access, which another server in the same subnet in the cloud provider could then exploit, if they were looking.

File: cloudproxy/check.py [hunk 2]
```diff
@@ -44,7 +44,13 @@ def fetch_ip(ip_address):
 
 def check_alive(ip_address):
     try:
-        result = requests.get("http://" + ip_address + ":8899", timeout=3)
+        proxies = {
+            "http": "http://" + ip_address + ":8899",
+            "https": "http://" + ip_address + ":8899"
+        }
+        result = requests.get("http://google.es", timeout=3, proxies=proxies)
```
Comment (claffin) [inline cloudproxy/check.py]: Risk of spamming Google with requests and Google may decide to block the IP, which would impact folks using this to scrape off Google.

I agree with the change but better to hit https://api.ipify.org instead as they expect this behavior.

**Review by claffin — CHANGES_REQUESTED** [review]

**Review by claffin — APPROVED** [review]

**Comment by claffin** [comment]
@serk7 reviewed and tested these look. I've added some questions about some of the changes you've made above. Let me know your thoughts otherwise I will make them myself as appreciate it's been a while since you submitted this PR.

**Comment by claffin** [comment]
Closed as stale, but will implement some features still and credit appropriately.

**Comment by serk7** [comment]
Hello Claffin,
About why i added /24. I really don't remember why i did this change to be honest.

**Comment by serk7** [comment]
Btw, i saw today the email about you closing it. For some reason i didn't read the other emails when you commented. I'm really sorry for don't response in time. Now i'm working but this evening i will check my commits and i will reply your questions.

**Pull request closed without merging**